Marlin/ultralcd_impl_DOGM.h

```diff
@@ -918,14 +918,15 @@ static void lcd_implementation_status_screen() {
       if (!PAGE_CONTAINS(row_y1, row_y2)) return;
 
       uint8_t n = LCD_WIDTH - (START_COL) - 1;
+      char c;
       if (longFilename[0]) {
         filename = longFilename;
-        longFilename[n] = '\0';
+        longFilename[n] = '\0'; // cutoff at screen edge
```

Comment:
I don't understand why this is necessary. The loop which draws the characters only draws one character at a time. And adding this '\0' here truncates whatever buffer longFilename points to, which is data that is not owned by this function. It could affect the long filename scrolling feature, if the buffer is reused and not refreshed.